### PR TITLE
Organize strategy tests folder

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

1. Consolidated all ddp test files into:
test_ddp.py: Unit tests
test_ddp_integration.py: Integration tests
2. Deduplicated/merged tests from test_ddp_strategy.py and test_ddp_spawn_strategy.py (these are no longer two seperate strategies as of 2.0)

No other changes apart from moving test functions around and renaming files.

cc @justusschock @awaelchli @borda

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: tests/tests_pytorch/strategies/test_ddp.py
Comment (Borda) [inline tests/tests_pytorch/strategies/test_ddp.py:1]: so let's rename it as `test_ddp_unittest.py` to be self-informative?

**Review by Borda — APPROVED** [review]

File: tests/tests_pytorch/strategies/test_ddp.py
Comment (awaelchli) [reply tests/tests_pytorch/strategies/test_ddp.py:1]: I prefer to keep it consistent with fabric, and renaming all files would be too much effort.

Comment (Borda) [reply tests/tests_pytorch/strategies/test_ddp.py:1]: rename just this one file... as it is storage one is integration so what the other one...

Comment (awaelchli) [reply tests/tests_pytorch/strategies/test_ddp.py:1]: I'd like to keep the name for consistency, both in Fabric and here.

**Pull request merged**